static/app/utils/profiling/renderers/internalSampleTickRenderer.tsx

```diff
@@ -0,0 +1,64 @@
+import {mat3} from 'gl-matrix';
+
+import {Flamegraph} from '../flamegraph';
+import {FlamegraphTheme} from '../flamegraph/flamegraphTheme';
+import {getContext, Rect} from '../gl/utils';
+
+function computeAbsoluteSampleTimestamps(startedAt: number, weights: readonly number[]) {
+  const timeline = [startedAt + weights[0]];
+  for (let i = 1; i < weights.length; i++) {
+    timeline.push(timeline[i - 1] + weights[i]);
+  }
+  return timeline;
+}
+
+class InternalSampleTickRenderer {
```

Comment:
It just occurred to me, but I would probably name this class `SampleTickRenderer`, the reason for that is that how the renderer is exposed to the user is not really it's "business" :D